### PR TITLE
update to 3.1 and fix ObjectProxy extends validations bug

### PR DESCRIPTION
**Pull request by danielspaniel**
Resolves an issue where you get this error:
 
Assertion Failed: You attempted to access the `__VALIDATIONS_MIXIN_COUNT__` property (of <(unknown mixin):ember391>).↵Since Ember 3.1, this is usually fine as you no longer need to use `.get()`↵to access computed properties. However, in this case, the object in question↵is a special kind of Ember object (a proxy). Therefore, it is still necessary↵to use `.get('__VALIDATIONS_MIXIN_COUNT__')`

when you try to extend ObjectProxy with validations and the content you are proxying is an ember-data model 

Changes proposed:

 - minor cleanup/update in ValidationsMixin within buildValidations to account for issues with extending ObjectProxy with Validations
- does not need documentation because it is merely fixing internal implementation
 
Tasks:

- [+ ] Added test case(s)

### CODE REVIEW

File: addon/validations/factory.js [hunk 1]
```diff
@@ -101,8 +101,8 @@ export default function buildValidations(validations = {}, globalOptions = {}) {
       this._super(...arguments);
 
       // Count number of mixins to bypass super check if there is more than 1
-      this[VALIDATIONS_MIXIN_COUNT] = this[VALIDATIONS_MIXIN_COUNT] || 0;
-      validationMixinCount = ++this[VALIDATIONS_MIXIN_COUNT];
+      validationMixinCount = (this.get(VALIDATIONS_MIXIN_COUNT) || 0) + 1;
```
Comment (offirgolan) [inline addon/validations/factory.js]: `get(this, VALIDATIONS_MIXIN_COUNT)`

File: addon/validations/factory.js [hunk 2]
```diff
@@ -101,8 +101,8 @@ export default function buildValidations(validations = {}, globalOptions = {}) {
...
+      this.set(VALIDATIONS_MIXIN_COUNT, validationMixinCount);
```
Comment (offirgolan) [inline addon/validations/factory.js]: `set(this, VALIDATIONS_MIXIN_COUNT, validationMixinCount)`

**Comment by offirgolan** [comment]
@danielspaniel thank you for this PR 😸. Added 2 small comments but other than that, lgtm!

**Comment by danielspaniel** [comment]
All set @offirgolan

**Comment by offirgolan** [comment]
Thanks again @danielspaniel!

**Comment by danielspaniel** [comment]
your welcome @offirgolan

**Comment by Turbo87** [comment]
@offirgolan would be open to backporting this fix to the v3.x branch?

**Pull request merged**